lib/semian/redis.rb

```diff
@@ -91,6 +91,30 @@ def connect
       end
     end
 
+    def with_resource_timeout(temp_timeout)
+      prev_timeouts = {
+        timeout: options[:timeout],
+        connect_timeout: options[:connect_timeout],
+        read_timeout: options[:read_timeout],
+        write_timeout: options[:write_timeout],
+      }
```

Comment:
Why not just use local variables?

Reply:
Didn't consider it. 💃 My internal heuristic "use hash when lots-o-variables" has failed me.